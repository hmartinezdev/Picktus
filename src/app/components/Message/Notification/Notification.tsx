import color from '@constants/colors';
import React, { Component } from 'react';

export interface INotificationProps {
  text: string;
  level: PicktusMessageDisplayType;
}

class Notification extends Component<INotificationProps> {
  public render(): React.ReactElement<Notification> {
    const { text, level } = this.props;
    return (
      <div>
        <div className={`notification notification--${level}`}>{text}</div>
        <style jsx>{`
          .notification {
            width: 200px;
            min-height: 100px;
            background-color: ${color.secondary};
            border-radius: 3px;
          }
        `}</style>
      </div>
    );
  }
}

export default Notification;
