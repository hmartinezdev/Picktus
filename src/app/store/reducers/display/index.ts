import display from './display-reducer';
export { IDisplayState } from './display.type';

export default display;
