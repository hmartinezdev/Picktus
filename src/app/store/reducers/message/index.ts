import message from './message-reducer';
export { IMessageState } from './message.type';

export default message;
