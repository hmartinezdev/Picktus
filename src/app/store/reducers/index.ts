import { combineReducers } from 'redux';
import display, { IDisplayState } from './display';
import message, { IMessageState } from './message';
import user, { IUserState } from './user';

export interface IReduxState {
  user: IUserState;
  display: IDisplayState;
  message: IMessageState;
}

export default combineReducers({
  display,
  message,
  user,
});
