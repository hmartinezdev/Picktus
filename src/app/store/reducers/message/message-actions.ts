import { ThunkAction } from 'redux-thunk';
import { IMessageState, IPicktusError, IPicktusMessage } from './message.type';

export type ThunkResult<R> = ThunkAction<R, IMessageState, undefined, DismissNotification>;

export enum TypeKeys {
  MESSAGE_DISMISS_NOTIFICATION = 'MESSAGE_DISMISS_NOTIFICATION',
  MESSAGE_DISMISS_SNACKBAR = 'MESSAGE_DISMISS_SNACKBAR',
  MESSAGE_DISPLAY_SNACKBAR = 'MESSAGE_DISPLAY_SNACKBAR',
  MESSAGE_DISPLAY_NOTIFICATION = 'MESSAGE_DISPLAY_NOTIFICATION',
}

export interface DisplaySnackBar {
  type: TypeKeys.MESSAGE_DISPLAY_SNACKBAR;
  message: IPicktusError | IPicktusMessage;
}

export const displaySnackBar = (message: IPicktusError | IPicktusMessage): DisplaySnackBar => ({
  message,
  type: TypeKeys.MESSAGE_DISPLAY_SNACKBAR,
});

export interface DisplayNotification {
  type: TypeKeys.MESSAGE_DISPLAY_NOTIFICATION;
  message: IPicktusError | IPicktusMessage;
}

export const DisplayNotification = (message: IPicktusError | IPicktusMessage): DisplayNotification => ({
  message,
  type: TypeKeys.MESSAGE_DISPLAY_NOTIFICATION,
});

export interface DismissNotification {
  type: TypeKeys.MESSAGE_DISMISS_NOTIFICATION;
}

export const dismissNotification = (): DismissNotification => ({
  type: TypeKeys.MESSAGE_DISMISS_NOTIFICATION,
});

export interface DismissSnackBar {
  type: TypeKeys.MESSAGE_DISMISS_SNACKBAR;
}

export const dismissSnackBar = (): DismissSnackBar => ({
  type: TypeKeys.MESSAGE_DISMISS_SNACKBAR,
});

export type MessageActions = DismissNotification | DismissSnackBar | DisplayNotification | DisplaySnackBar;
