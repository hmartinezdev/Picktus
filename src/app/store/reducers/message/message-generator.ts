import uuid from 'uuid';
import { IPicktusError, IPicktusMessage, PicktusMessageDisplay, PicktusMessageLevel } from './message.type';

export default function(
  text: string,
  level: PicktusMessageLevel,
  display: PicktusMessageDisplay
): IPicktusError | IPicktusMessage {
  return {
    display,
    id: uuid.v4(),
    level,
    text,
  };
}
