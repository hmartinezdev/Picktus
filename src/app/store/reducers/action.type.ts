import { DisplayActions } from './display/display-actions';
import { MessageActions } from './message/message-actions';
import { UserActions } from './user/user-actions';

enum TypeKeys {
  OTHER_ACTION = '__any_other_action_type__',
}

interface OtherAction {
  type: TypeKeys.OTHER_ACTION;
}

export type ActionTypes = UserActions | DisplayActions | MessageActions | OtherAction;
