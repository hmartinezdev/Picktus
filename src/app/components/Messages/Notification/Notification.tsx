import colors from '@constants/colors';
import { borderRadius, boxShadow, fontFamily } from '@constants/styles';
import { PicktusMessageLevel } from '@store/reducers/message';
import React, { Component } from 'react';
import MessageIcon from '../MessageIcon';

export interface INotificationProps {
  text: string;
  level: PicktusMessageLevel;
}

class Notification extends Component<INotificationProps> {
  public render(): React.ReactElement<Notification> {
    const { text, level } = this.props;
    return (
      <div>
        <div className={`notification notification--${level}`}>
          <div className="iconContainer">
            <MessageIcon level={level} />
          </div>
          <p>{text}</p>
        </div>
        <style jsx>{`
          .notification {
            display: flex;
            align-items: center;
            box-sizing: border-box;
            padding: 1rem;
            font-size: 1.05rem;
            font-family: ${fontFamily};
            color: ${colors.white};
            width: 100%;
            min-height: 5rem;
            background-color: ${colors.primary};
            border-radius: ${borderRadius};
            box-shadow: ${boxShadow};
          }

          .notification--${PicktusMessageLevel.ERROR} {
            background-color: ${colors.error};
          }

          .notification--${PicktusMessageLevel.SUCCESS} {
            background-color: ${colors.green};
          }

          .notification--${PicktusMessageLevel.WARNING} {
            background-color: ${colors.warning};
          }

          .iconContainer {
            min-width: 32px;
            margin-right: 0.6rem;
          }
        `}</style>
      </div>
    );
  }
}

export default Notification;
