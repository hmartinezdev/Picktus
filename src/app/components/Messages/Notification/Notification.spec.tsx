import { PicktusMessageLevel } from '@store/reducers/message';
import { shallow } from 'enzyme';
import React from 'react';
import Notification from './Notification';

let wrapper;
const baseProps = { text: 'test', level: PicktusMessageLevel.INFO };

const setup = (props = baseProps) => shallow(<Notification {...props} />);

describe('<Notification />', () => {
  beforeEach(() => {
    wrapper = setup();
  });

  test('should render properly', () => {
    wrapper = setup();
    expect(wrapper).toMatchSnapshot();
  });

  test('should render with different icon', () => {
    wrapper = setup({ level: PicktusMessageLevel.ERROR, text: 'error' });
    expect(wrapper).toMatchSnapshot();
    wrapper = setup({ level: PicktusMessageLevel.SUCCESS, text: 'error' });
    expect(wrapper).toMatchSnapshot();
    wrapper = setup({ level: PicktusMessageLevel.WARNING, text: 'error' });
    expect(wrapper).toMatchSnapshot();
    wrapper = setup({ level: PicktusMessageLevel.INFO, text: 'error' });
    expect(wrapper).toMatchSnapshot();
  });
});
