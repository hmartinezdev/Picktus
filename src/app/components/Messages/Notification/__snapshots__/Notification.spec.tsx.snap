// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Notification /> should render properly 1`] = `
<div>
  <div
    className="notification notification--INFO"
  >
    <div
      className="iconContainer"
    >
      <MessageIcon
        height={1.8}
        level="INFO"
        width={1.8}
      />
    </div>
    <p>
      test
    </p>
  </div>
  <style
    jsx={true}
  >
    
          .notification {
            display: flex;
            align-items: center;
            box-sizing: border-box;
            padding: 1rem;
            font-size: 1.05rem;
            font-family: 'Josefin Sans', sans-serif;
            color: #F8F8F7;
            width: 100%;
            min-height: 5rem;
            background-color: #2d3047;
            border-radius: 3px;
            box-shadow: 11px 10px 34px -7px rgba(0, 0, 0, 0.49);
          }

          .notification--ERROR {
            background-color: #982649;
          }

          .notification--SUCCESS {
            background-color: #75ab88;
          }

          .notification--WARNING {
            background-color: #E37E5B;
          }

          .iconContainer {
            min-width: 32px;
            margin-right: 0.6rem;
          }
        
  </style>
</div>
`;

exports[`<Notification /> should render with different icon 1`] = `
<div>
  <div
    className="notification notification--ERROR"
  >
    <div
      className="iconContainer"
    >
      <MessageIcon
        height={1.8}
        level="ERROR"
        width={1.8}
      />
    </div>
    <p>
      error
    </p>
  </div>
  <style
    jsx={true}
  >
    
          .notification {
            display: flex;
            align-items: center;
            box-sizing: border-box;
            padding: 1rem;
            font-size: 1.05rem;
            font-family: 'Josefin Sans', sans-serif;
            color: #F8F8F7;
            width: 100%;
            min-height: 5rem;
            background-color: #2d3047;
            border-radius: 3px;
            box-shadow: 11px 10px 34px -7px rgba(0, 0, 0, 0.49);
          }

          .notification--ERROR {
            background-color: #982649;
          }

          .notification--SUCCESS {
            background-color: #75ab88;
          }

          .notification--WARNING {
            background-color: #E37E5B;
          }

          .iconContainer {
            min-width: 32px;
            margin-right: 0.6rem;
          }
        
  </style>
</div>
`;

exports[`<Notification /> should render with different icon 2`] = `
<div>
  <div
    className="notification notification--SUCCESS"
  >
    <div
      className="iconContainer"
    >
      <MessageIcon
        height={1.8}
        level="SUCCESS"
        width={1.8}
      />
    </div>
    <p>
      error
    </p>
  </div>
  <style
    jsx={true}
  >
    
          .notification {
            display: flex;
            align-items: center;
            box-sizing: border-box;
            padding: 1rem;
            font-size: 1.05rem;
            font-family: 'Josefin Sans', sans-serif;
            color: #F8F8F7;
            width: 100%;
            min-height: 5rem;
            background-color: #2d3047;
            border-radius: 3px;
            box-shadow: 11px 10px 34px -7px rgba(0, 0, 0, 0.49);
          }

          .notification--ERROR {
            background-color: #982649;
          }

          .notification--SUCCESS {
            background-color: #75ab88;
          }

          .notification--WARNING {
            background-color: #E37E5B;
          }

          .iconContainer {
            min-width: 32px;
            margin-right: 0.6rem;
          }
        
  </style>
</div>
`;

exports[`<Notification /> should render with different icon 3`] = `
<div>
  <div
    className="notification notification--WARNING"
  >
    <div
      className="iconContainer"
    >
      <MessageIcon
        height={1.8}
        level="WARNING"
        width={1.8}
      />
    </div>
    <p>
      error
    </p>
  </div>
  <style
    jsx={true}
  >
    
          .notification {
            display: flex;
            align-items: center;
            box-sizing: border-box;
            padding: 1rem;
            font-size: 1.05rem;
            font-family: 'Josefin Sans', sans-serif;
            color: #F8F8F7;
            width: 100%;
            min-height: 5rem;
            background-color: #2d3047;
            border-radius: 3px;
            box-shadow: 11px 10px 34px -7px rgba(0, 0, 0, 0.49);
          }

          .notification--ERROR {
            background-color: #982649;
          }

          .notification--SUCCESS {
            background-color: #75ab88;
          }

          .notification--WARNING {
            background-color: #E37E5B;
          }

          .iconContainer {
            min-width: 32px;
            margin-right: 0.6rem;
          }
        
  </style>
</div>
`;

exports[`<Notification /> should render with different icon 4`] = `
<div>
  <div
    className="notification notification--INFO"
  >
    <div
      className="iconContainer"
    >
      <MessageIcon
        height={1.8}
        level="INFO"
        width={1.8}
      />
    </div>
    <p>
      error
    </p>
  </div>
  <style
    jsx={true}
  >
    
          .notification {
            display: flex;
            align-items: center;
            box-sizing: border-box;
            padding: 1rem;
            font-size: 1.05rem;
            font-family: 'Josefin Sans', sans-serif;
            color: #F8F8F7;
            width: 100%;
            min-height: 5rem;
            background-color: #2d3047;
            border-radius: 3px;
            box-shadow: 11px 10px 34px -7px rgba(0, 0, 0, 0.49);
          }

          .notification--ERROR {
            background-color: #982649;
          }

          .notification--SUCCESS {
            background-color: #75ab88;
          }

          .notification--WARNING {
            background-color: #E37E5B;
          }

          .iconContainer {
            min-width: 32px;
            margin-right: 0.6rem;
          }
        
  </style>
</div>
`;
